main/FatesHistoryInterfaceMod.F90

```diff
@@ -5240,6 +5252,160 @@ end subroutine update_history_hifrq_sitelevel
+    integer                 , intent(in)            :: nsites
+    type(ed_site_type)      , intent(inout), target :: sites(nsites)
+    type(bc_in_type)        , intent(in)            :: bc_in(nsites)
+    real(r8)                , intent(in)            :: dt_tstep
+
+    ! Locals
+    integer  :: s        ! The local site index
+    integer  :: io_si     ! The site index of the IO array
+
+    real(r8) :: landuse_statevector(n_landuse_cats)
+    real(r8) :: canopy_area_bylanduse(n_landuse_cats)
+    integer  :: i_lu
+    logical  :: foundbaregroundpatch
+
+    type(fates_patch_type),pointer  :: cpatch
+    type(fates_cohort_type),pointer :: ccohort
+    real(r8) :: dt_tstep_inv          ! Time step in frequency units (/s)
+
+    associate( hio_tveg_si_landuse    => this%hvars(ih_tveg_si_landuse)%r82d,&
+         hio_gpp_si_landuse           => this%hvars(ih_gpp_si_landuse)%r82d, &
+         hio_tsa_si_landuse           => this%hvars(ih_tsa_si_landuse)%r82d,&
+         hio_sw_abs_si_landuse        => this%hvars(ih_sw_abs_si_landuse)%r82d,&
+         hio_lw_net_si_landuse        => this%hvars(ih_lw_net_si_landuse)%r82d,&
+         hio_shflux_si_landuse        => this%hvars(ih_shflux_si_landuse)%r82d,&
+         hio_lhflux_si_landuse        => this%hvars(ih_lhflux_si_landuse)%r82d)
+
+      do_sites: do s = 1,nsites
+
+         io_si  = sites(s)%h_gid
+
+         dt_tstep_inv = 1.0_r8/dt_tstep
+
+         ! biophysical properties that are indexed by land use
+         landuse_statevector(:) = sites(s)%get_current_landuse_statevector() * AREA
+
+         ! get the total canopy area for each land use type
+         canopy_area_bylanduse(:) = 0._r8
+         cpatch => sites(s)%oldest_patch
+         do while(associated(cpatch))
+            if ( cpatch%land_use_label .ne. nocomp_bareground_land ) then
+               canopy_area_bylanduse(cpatch%land_use_label) = canopy_area_bylanduse(cpatch%land_use_label) + &
+                    cpatch%total_canopy_area
+            endif
+            cpatch => cpatch%younger
+         end do
+
+         cpatch => sites(s)%oldest_patch
+         do while(associated(cpatch))
+            if (cpatch%total_canopy_area .gt. rsnbl_math_prec) then
+               ! for TVEG, since it is only defined on vegetated area of vegetated patches, normalize by the total vegetated area
+               hio_tveg_si_landuse(io_si,cpatch%land_use_label) = hio_tveg_si_landuse(io_si,cpatch%land_use_label) + &
+                    bc_in(s)%t_veg_pa(cpatch%patchno) * cpatch%total_canopy_area/canopy_area_bylanduse(cpatch%land_use_label)
+
+               ! for the rest of these, first weight by the vegetated area of each patch over the total patch area for each land use type
+               hio_tsa_si_landuse(io_si,cpatch%land_use_label) = hio_tsa_si_landuse(io_si,cpatch%land_use_label) + &
+                    bc_in(s)%t2m_pa(cpatch%patchno) * cpatch%total_canopy_area/landuse_statevector(cpatch%land_use_label)
```

Comment:
For normalization factors like `cpatch%total_canopy_area/landuse_statevector(cpatch%land_use_label)` that are repeated a bunch, it'd be good to create a variable for them with a useful name. E.g., what I did in `update_history_dyn_subsite_ageclass()` with [`patch_area_div_site_area`](https://github.com/NGEET/fates/blob/fe782ba4f851f8c8bfac8a89607e7c05bd397553/main/FatesHistoryInterfaceMod.F90#L4751).